### PR TITLE
find peptide formula

### PR DESCRIPTION
**Pull request by kyp4**
calculates the full peptide formula with mods

### CODE REVIEW

File: FlashLFQ/Peptide.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using System.Collections.Generic;
+﻿using Chemistry;
```
Comment (trishorts) [inline FlashLFQ/Peptide.cs:1]: unused, please remove

File: Test/TestDeconvolution.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void TestDeconvolutionProteoformMultiChargeState(double selectedIonMz, in
 
             MsDataScan scan = myMsDataFile.GetAllScansList()[0];
 
-            List<IsotopicEnvelope> isolatedMasses = scan.GetIsolatedMassesAndCharges(spectrum, 1, 60, 4, 3).ToList();
+            List<MassSpectrometry.IsotopicEnvelope> isolatedMasses = scan.GetIsolatedMassesAndCharges(spectrum, 1, 60, 4, 3).ToList();
```
Comment (trishorts) [inline Test/TestDeconvolution.cs:50]: this needs an assertion

File: Test/TestPeptideWithSetMods.cs [hunk 3]
```diff
@@ -929,5 +930,38 @@ public static void CountTargetsWithMatchingDecoys()
                 }
             }
         }
+
+
+        [Test]
+        public static void CheckFullChemicalFormula()
+        {
+            PeptideWithSetModifications small_pep = new PeptideWithSetModifications(new Protein("PEPTIDE", "ASSESSION"), new DigestionParams(protease: "trypsin"), 1, 7, CleavageSpecificity.Full, null, 0, new Dictionary<int, Modification>(), 0, null);
```
Comment (trishorts) [inline Test/TestPeptideWithSetMods.cs:938]: spelling

File: Test/TestPeptideWithSetMods.cs [hunk 4]
```diff
@@ -929,5 +930,38 @@ public static void CountTargetsWithMatchingDecoys()
...
+            ChemicalFormula small_pep_cf = ChemicalFormula.ParseFormula("C34H53N7O15");
+            Assert.AreEqual(small_pep.FullChemicalFormula, small_pep_cf);
+
+            PeptideWithSetModifications large_pep = new PeptideWithSetModifications(new Protein("PEPTIDEKRNSPEPTIDEKECUEIRQUV", "ASSESSION"), new DigestionParams(protease: "trypsin"), 1, 28, CleavageSpecificity.Full, null, 0, new Dictionary<int, Modification>(), 0, null);
```
Comment (trishorts) [inline Test/TestPeptideWithSetMods.cs:942]: spelling

File: Test/TestPeptideWithSetMods.cs [hunk 5]
```diff
@@ -929,5 +930,38 @@ public static void CountTargetsWithMatchingDecoys()
...
+            ChemicalFormula large_pep_cf = ChemicalFormula.ParseFormula("C134H220N38O50S1Se2");
+            Assert.AreEqual(large_pep.FullChemicalFormula, large_pep_cf);
+
+            ModificationMotif.TryGetMotif("S", out ModificationMotif motif_s);
+            Modification phosphorylation = new Modification(_originalId: "phospho", _modificationType: "CommonBiological", _target: motif_s, _locationRestriction: "Anywhere.", _chemicalFormula: ChemicalFormula.ParseFormula("H1O3P1"));
+            Dictionary<int, Modification> modDict_small = new Dictionary<int, Modification>();
+            modDict_small.Add(4, phosphorylation);
+
+            PeptideWithSetModifications small_pep_mod = new PeptideWithSetModifications(new Protein("PEPSIDE", "ASSESSION"), new DigestionParams(protease: "trypsin"), 1, 7, CleavageSpecificity.Full, null, 0, modDict_small, 0, null);
```
Comment (trishorts) [inline Test/TestPeptideWithSetMods.cs:951]: spelling

**Review by trishorts — CHANGES_REQUESTED** [review]
looking better. just a couple more little items to deal with

File: Test/TestDeconvolution.cs (hunk 2 above)
Comment (kyp4) [reply Test/TestDeconvolution.cs:50]: Changed test back to its original form, so there should be no need to add an assertion

**Review by acesnik — DISMISSED** [review]
Looks good to me.

**Review by Alexander-Sol — APPROVED** [review]
10/10 code, would review again

**Review by nbollis — APPROVED** [review]
This code has been blessed

**Review by trishorts — APPROVED** [review]
looks good.

**Review by nbollis — APPROVED** [review]

**Pull request merged**